src/iamService.ts

```diff
@@ -0,0 +1,383 @@
+      this._opts
+    ) as Promise<{[method: string]: Function}>;
+    // Iterate over each of the methods that the service provides
+    // and create an API call method for each.
+    const iamPolicyStubMethods = [
+      'getIamPolicy',
+      'setIamPolicy',
+      'testIamPermissions',
+    ];
+
+    for (const methodName of iamPolicyStubMethods) {
+      const innerCallPromise = this.iamPolicyStub.then(
+        stub => (...args: Array<{}>) => {
+          if (this._terminated) {
+            return Promise.reject('The client has already been closed.');
+          }
+          const func = stub[methodName];
+          return func.apply(stub, args);
+        },
+        (err: Error | null | undefined) => () => {
+          throw err;
+        }
+      );
+      this.innerApiCalls[methodName] = createApiCall(
+        innerCallPromise,
+        this._defaults[methodName],
+        this.descriptors.page[methodName]
+      );
+    }
+    return this.iamPolicyStub;
+  }
+
+  /**
+   * The DNS address for this API service.
+   */
+  static get servicePath() {
+    return 'cloudkms.googleapis.com';
+  }
+
+  /**
+   * The DNS address for this API service - same as servicePath(),
+   * exists for compatibility reasons.
+   */
+  static get apiEndpoint() {
+    return 'cloudkms.googleapis.com';
+  }
+
+  /**
+   * The port for this API service.
+   */
+  static get port() {
+    return 443;
+  }
+  /**
+   * The scopes needed to make gRPC calls for every method defined
+   * in this service.
+   */
+  static get scopes() {
+    return [
+      'https://www.googleapis.com/auth/cloud-platform',
+      'https://www.googleapis.com/auth/cloudkms',
+    ];
+  }
+
+  /**
+   * Get the project ID used by this class.
+   * @param {function(Error, string)} callback - the callback to be called with
+   *   the current project Id.
+   */
+  getProjectId(): Promise<string>;
+  getProjectId(callback: ProjectIdCallback): void;
+  getProjectId(callback?: ProjectIdCallback): void | Promise<string> {
+    if (this.auth && 'getProjectId' in this.auth) {
+      return this.auth.getProjectId(callback!);
+    }
+    if (callback) {
+      callback(new Error('Cannot determine project ID.'));
+    } else {
+      return Promise.reject('Cannot determine project ID.');
+    }
+  }
+
+  getIamPolicy(
+    request: protos.google.iam.v1.GetIamPolicyRequest,
+    options?: gax.CallOptions
+  ): Promise<protos.google.iam.v1.Policy>;
+  getIamPolicy(
+    request: protos.google.iam.v1.GetIamPolicyRequest,
+    options: gax.CallOptions,
+    callback: protos.google.iam.v1.IAMPolicy.GetIamPolicyCallback
+  ): void;
+  getIamPolicy(
+    request: protos.google.iam.v1.GetIamPolicyRequest,
+    callback: protos.google.iam.v1.IAMPolicy.GetIamPolicyCallback
+  ): void;
+  getIamPolicy(
+    request: protos.google.iam.v1.GetIamPolicyRequest,
+    optionsOrCallback?:
+      | gax.CallOptions
+      | protos.google.iam.v1.IAMPolicy.GetIamPolicyCallback,
```

Comment:
What is `protos.google.iam.v1.IAMPolicy.GetIamPolicyCallback`?

Reply:
I mean, should it be `Callback<......>` instead?

Reply:
done.